configure.ac

```diff
@@ -127,16 +127,34 @@ fi
 #---
 # Check for ICU
 #---
+HAVE_ICU=0
 AC_ARG_WITH([icu],
   AS_HELP_STRING([--without-icu],
     [Disable Internation Components for Unicode. WARNING: Many components will stop working without ICU support!]), [], [with_icu=yes])
-AS_IF([test x"with_icu" != xno],
-  [PKG_CHECK_MODULES([ICU], [icu-i18n > 49.0])
-  AC_CHECK_HEADERS([unicode/ucal.h unicode/uchar.h unicode/ucnv.h unicode/ucol.h unicode/ucurr.h unicode/udat.h unicode/udatpg.h unicode/uloc.h unicode/ulocdata.h unicode/unorm.h unicode/unum.h unicode/usearch.h unicode/ustring.h unicode/utrans.h],
-  [], AC_MSG_ERROR([Could not find required ICU headers.]))
+AS_IF([test x"$with_icu" != xno], [
```

Comment:
I feel there should be a way to simplify this check. We really only use PKG_CHECK_MODULES to check the version of the installed ICU library. I don't really know how ICU is packaged in Windows, but it sounds like only icu.h is provided. Maybe this check for icu.h can be done directly in AC_CHECK_HEADERS's "action-if-not-found" parameter.

Reply:
I’ve tried to simplify it as much as I could. It’s a bit complex because on Windows we need to try using the bundled ICU first in order to avoid running PKG_CHECK_MODULES and getting an error, but we want to fall back to pkg-config if the bundled ICU is not found (on older Windows releases). Please let me know if you have any suggestions on how to simplify this further.